pandas/core/groupby/generic.py

```diff
@@ -1100,13 +1100,19 @@ def py_fallback(bvalues: ArrayLike) -> ArrayLike:
             #  in the operation.  We un-split here.
             result = result._consolidate()
             assert isinstance(result, (Series, DataFrame))  # for mypy
-            mgr = result._mgr
-            assert isinstance(mgr, BlockManager)
-            assert len(mgr.blocks) == 1
 
             # unwrap DataFrame to get array
-            result = mgr.blocks[0].values
-            return result
+            mgr = result._mgr
```

Comment:
instead of this can you now just define as_array properly?

Reply:
of simply consolidate first?

Reply:
It's already consolidated above, but if you have multiple dtypes, that doesn't help

Reply:
To be clear, as the diff seems to be larger (because I remove the assert above and moved the comment), but basically the only code addition is this:

```
            if len(mgr.blocks) != 1:
                return mgr.as_array()
```

to handle the case of multiple blocks that need to be converted into an array (while now this raised an AssertionError because of `assert len(mgr.blocks) == 1`